src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationStrategy {
+
+    @Override
+    public void processTransaction(FruitTransaction transaction, Storage storage) {
+        int currentQuantity = storage.getFruitCount(transaction.getFruit());
+        int resultQuantity = currentQuantity - transaction.getQuantity();

```

Comment:
any check that we don't have negative value as a result?

Reply:
fixed